### ISSUE TITLE
Sysrepo installation error on Buildroot 

### ISSUE THREAD

**Issue by antonioamjr** [sysrepo/sysrepo#1498]
### Description
I'm trying to add sysrepo to my buildroot image for a project and it has been hard to get it accomplished.  
I noticed in the [install doc](https://github.com/sysrepo/sysrepo/blob/master/INSTALL.md#cross-compiling-notes) that the binaries produced on build are still used in a later step, so I used the last approach mentioned on the cross-compiling section to disable `CALL_TARGET_BINS_DIRECTLY`and install other time with the install-yang.sh script.  
The build then works, but I still facing problems installing to the staging directory (that I think it's necessary since I'll be adding netopeer2 package next). Am I missing something?

### Steps to Reproduce
1. Load the buildroot config from configs/qemu_x86_64_defconfig
2. Select protobuf, protobuf-c, pcre, libev, python3, host-swig and cmocka packages from the buildroot menuconfig
3. Add buildroot new packages libyang and libredblack and select them in buildroot menuconfig
4. Run Buildroot's `make`

**Expected behavior:**
 Successfully install the sysrepo by buildroot's build system.

**Actual behavior:**
 Installing to staging directory step fails when fixing libtool files hook is called. It shows the following log (Verbose mode):
```
>>> sysrepo v0.7.7 Fixing libtool files
find /home/aamjr/dev/buildroot-2018.02.1/output/host/x86_64-buildroot-linux-uclibc/sysroot/usr/lib* -name "*.la" | xargs --no-run-if-empty \
	/bin/sed -i -e "s:/home/host/dev/buildroot-2018.02.1/output:@BASE_DIR@:g" \
		-e "s:/home/host/dev/buildroot-2018.02.1/output/host/x86_64-buildroot-linux-uclibc/sysroot:@STAGING_DIR@:g" \
		 \
		-e "s:\(['= ]\)/usr:\\1@STAGING_DIR@/usr:g" \
		 \
		-e "s:@STAGING_DIR@:/home/host/dev/buildroot-2018.02.1/output/host/x86_64-buildroot-linux-uclibc/sysroot:g" \
		-e "s:@BASE_DIR@:/home/host/dev/buildroot-2018.02.1/output:g"
sysrepo: installs files in /home/host/dev/buildroot-2018.02.1/output/host/x86_64-buildroot-linux-uclibc/sysroot//home/host/dev/buildroot-2018.02.1/output
package/pkg-generic.mk:260: recipe for target '/home/host/dev/buildroot-2018.02.1/output/build/sysrepo-v0.7.7/.stamp_staging_installed' failed
make: *** [/home/host/dev/buildroot-2018.02.1/output/build/sysrepo-v0.7.7/.stamp_staging_installed] Error 1
```

### Versions
Buildroot version: 2018.02.1
Sysrepo version: v0.7.7
libyang version: v0.16-r2

### Files
Buildroot's Config.in file content: 
```
config BR2_PACKAGE_SYSREPO
	bool "sysrepo"
	depends on BR2_PACKAGE_LIBEV
	depends on BR2_PACKAGE_LIBREDBLACK
	depends on BR2_PACKAGE_LIBYANG
	depends on BR2_PACKAGE_PROTOBUF_C
	depends on !BR2_STATIC_LIBS
	depends on BR2_TOOLCHAIN_HAS_THREADS
        depends on BR2_INSTALL_LIBSTDCPP
	help
	  Sysrepo is a YANG-based configuration and operational
          state data store for Unix/Linux applications.
          http://www.sysrepo.org/
```

Buildroot's sysrepo.mk file content:
```
#########################################################################
#sysrepo
##########################################################################

SYSREPO_VERSION = v0.7.7
SYSREPO_SOURCE = $(SYSREPO_VERSION).tar.gz
SYSREPO_SITE = https://github.com/sysrepo/sysrepo/archive
SYSREPO_LICENSE = Apache-2.0
SYSREPO_LICENSE_FILES = LICENSE
SYSREPO_INSTALL_STAGING = YES

SYSREPO_CONF_OPTS += \
	-DCMAKE_BUILD_TYPE=Release \
	-DBUILD_EXAMPLES:BOOL=FALSE \
	-DBUILD_CPP_EXAMPLES:BOOL=0 \
	-DCALL_TARGET_BINS_DIRECTLY:BOOL=OFF \
	-DENABLE_TESTS=FALSE

$(eval $(cmake-package))
```

Thanks and kind regards!

**Comment by hthiery** [comment]
Hi,

i'm currently working on the integration of sysrepo (and dependencies) into mainline buildroot.

Just now these packages are accepted and integratet in the master branch. But I still struggle with some strange behaviors for the sysrepo package.

My issue currently is that (as far as I understand at the moment) the host build needs to set the REPOSITORY_LOC for the later target build. But wiht setting this variable also other stuff is installed into the target direcotry from the host build.

**Comment by jktjkt** [comment]
Here's the packaging that I've been using for the last year or two, IIRC:

- [sysrepo](https://gerrit.cesnet.cz/plugins/gitiles/github/buildroot/buildroot/+/refs/heads/cesnet/2019-06-10/package/sysrepo/)
- [netopeer2-server](https://gerrit.cesnet.cz/plugins/gitiles/github/buildroot/buildroot/+/refs/heads/cesnet/2019-06-10/package/netopeer2-server)

The other usual suspects (libyang, libnetconf2, keystored, etc) are also there.

The TL;DR version for the **host** build of sysrepo is `HOST_SYSREPO_CONF_OPTS += -DREPOSITORY_LOC=$(BASE_TARGET_DIR)/etc/sysrepo` so that whatever gets done by host's `sysrepoctl` and `sysrepocfg` ends up on target's `/etc/sysrepo`.

**Comment by hthiery** [comment]
Hi Jan,

> * [sysrepo](https://gerrit.cesnet.cz/plugins/gitiles/github/buildroot/buildroot/+/refs/heads/cesnet/2019-06-10/package/sysrepo/)
> * [netopeer2-server](https://gerrit.cesnet.cz/plugins/gitiles/github/buildroot/buildroot/+/refs/heads/cesnet/2019-06-10/package/netopeer2-server)

As you maybe know I work on a mainline integration for this buildroot packages.

> The TL;DR version for the **host** build of sysrepo is `HOST_SYSREPO_CONF_OPTS += -DREPOSITORY_LOC=$(BASE_TARGET_DIR)/etc/sysrepo` so that whatever gets done by host's `sysrepoctl` and `sysrepocfg` ends up on target's `/etc/sysrepo`.

The fact that the host build generates results that ends up in the target's directory is a no-go for the buildroot maintainers. Therefore I'm looking for a way to fix that issue in a more elegant way.

What about building only the sysrepocfg and sysrepoctl binary in the host build? What do you think?

**Comment by jktjkt** [comment]
> The fact that the host build generates results that ends up in the target's directory is a no-go for the buildroot maintainers. Therefore I'm looking for a way to fix that issue in a more elegant way.

I don't follow -- the host build always happens first, which means that it produces a repo content which then gets overwritten by the target build. Then, as we start building other target packages which require host's SR binaries, these packages install YANG Modules and enable features which will affect the resulting image. That's by design.

How do you and/or the other buildroot maintainers expect this to work?

> What about building only the sysrepocfg and sysrepoctl binary in the host build? What do you think?

You can probably patch these sources to skip building the daemons, but apart from some wasted CPU time, I wonder why bother. This really builds quickly enough, IMHO.

> My issue currently is that (as far as I understand at the moment) the host build needs to set the REPOSITORY_LOC for the later target build. But wiht setting this variable also other stuff is installed into the target direcotry from the host build.

Can you specify what exactly gets installed, and why is that a problem? I can imagine that test YANG modules needed for SR's tests would be a nuisance, but given that buildroot explicitly skips tests, I think I must be missing something.

**Comment by hthiery** [comment]
> > The fact that the host build generates results that ends up in the target's directory is a no-go for the buildroot maintainers. Therefore I'm looking for a way to fix that issue in a more elegant way.
> 
> I don't follow -- the host build always happens first, which means that it produces a repo content which then gets overwritten by the target build. Then, as we start building other target packages which require host's SR binaries, these packages install YANG Modules and enable features which will affect the resulting image. That's by design.
> 
> How do you and/or the other buildroot maintainers expect this to work?

you can see that the comment from Thomas was that it is not allowed to install in the host step into the target directory.
http://lists.busybox.net/pipermail/buildroot/2019-October/263853.html

> > What about building only the sysrepocfg and sysrepoctl binary in the host build? What do you think?
> 
> You can probably patch these sources to skip building the daemons, but apart from some wasted CPU time, I wonder why bother. This really builds quickly enough, IMHO.

I think it is not required to patch it. I tried to use a "custom" buildroot build and install command for the host:
```
HOST_SYSREPO_CONF_OPTS = \
	-DREPOSITORY_LOC=$(TARGET_DIR)/etc/sysrepo \

define HOST_SYSREPO_BUILD_CMDS
	warning dafs
	$(HOST_MAKE_ENV) $(MAKE) $(HOST_CONFIGURE_OPTS) -C $(@D) \
		sysrepocfg sysrepoctl
endef

define HOST_SYSREPO_INSTALL_CMDS
    $(INSTALL) -D -m 755 $(@D)/src/sysrepoctl $(HOST_DIR)/bin/sysrepoctl
    $(INSTALL) -D -m 755 $(@D)/src/sysrepocfg $(HOST_DIR)/bin/sysrepocfg
endef
```
this should not install anything else than the required SR executables for the target build


> > My issue currently is that (as far as I understand at the moment) the host build needs to set the REPOSITORY_LOC for the later target build. But wiht setting this variable also other stuff is installed into the target direcotry from the host build.
> 
> Can you specify what exactly gets installed, and why is that a problem? I can imagine that test YANG modules needed for SR's tests would be a nuisance, but given that buildroot explicitly skips tests, I think I must be missing something.

**Comment by jktjkt** [comment]
What @tpetazzoni said is that you are not supposed to install Python stuff into your *system*'s `/usr/lib/python2.7/site-packages/`. It does not say anything about a shared sysrepo location.

**Comment by hthiery** [comment]
> What @tpetazzoni said is that you are not supposed to install Python stuff into your _system_'s `/usr/lib/python2.7/site-packages/`. It does not say anything about a shared sysrepo location.

OK .. you're right. That was the not allowed. My fault.

but what about the thing with "Another issue is that the host-sysrepo installation installs stuff to
$(TARGET_DIR) which really isn't good for a host package:"

**Comment by hthiery** [comment]
Hi Jan,

I think the problem is solved. My fault was that the IS_DEVELOPERS_CONFIGURATION was set. Sorry for bothering you.

**Comment by hthiery** [comment]
> Hi Jan,
> 
> I think the problem is solved. My fault was that the IS_DEVELOPERS_CONFIGURATION was set. Sorry for bothering you.

sorry to come back .. but this was not the final solution.

**Comment by kishore-vvdn** [comment]
facing same issue can anyone help on this issue
>>> sysrepo v0.7.8 Fixing libtool files
sysrepo: installs files in /home/kishore/PROJECTS/STLI_SCEL/output/host/aarch64-buildroot-linux-gnu/sysroot//home/kishore/PROJECTS/STLI_SCEL/output
make[1]: *** [package/pkg-generic.mk:276: /home/kishore/PROJECTS/STLI_SCEL/output/build/sysrepo-v0.7.8/.stamp_staging_installed] Error 1
make: *** [Makefile:23: _all] Error 2

thanks
